### PR TITLE
mailnotification: add image inline as attachment; fix #2869

### LINKED ISSUE #2869: [9.2] image insertion in mail ticket notification
Since the 9.2 upgrade, when we receive a mail notification on a ticket update,
the images linked in the ticket do not appear like they used to do : 

Here is the result we get on the mail :
image
It seems the link to the docid is not complete.

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2869

### CODE REVIEW

File: inc/notificationeventmailing.class.php [hunk 1]
```diff
@@ -154,14 +157,52 @@ static public function send(array $data) {
...
+                        }
+                     }
                   }
                }
             }
+
+            // replace img[src] and a[href] by cid:tag in html content
+            foreach($docs_attached as $docID => $tag) {
+               $current->fields['body_html']
+                  = preg_replace('/src=["\'].*document\.send\.php\?docid='.$docID.'["\']/',
```
Comment (trasher) [inline inc/notificationeventmailing.class.php]: Only one preg_replace with arrays for pattern and replacement would be enough

File: inc/document.class.php [hunk 2]
```diff
@@ -460,11 +460,14 @@ static function getMaxUploadSize() {
 
    /**
     * Send a document to navigator
+    *
+    * @param string $context COntext to resize image, if any
```
Comment (btry) [inline inc/document.class.php]: typo

File: inc/document.class.php [hunk 3]
```diff
@@ -1416,4 +1419,82 @@ static function getImageTag($string) {
...
+            case 'timeline':
+               $mwidth = 100;
+               $mheight = 100;
+               break;
+            default:
+               throw new \RuntimeException("Unknown context $context!");
+         }
+      }
+
+      //let's see if originale image needs resize
```
Comment (btry) [inline inc/document.class.php]: original

**Comment by orthagh** [comment]
With last modification, seems ok for merge

**Pull request merged**